scripts/common_windows.sh

```diff
@@ -0,0 +1,130 @@
+  while read -r version; do
+    if [[ "$AWS_FOR_FLUENT_BIT_LATEST_VERSION" < "$version" ]]; then
+      AWS_FOR_FLUENT_BIT_LATEST_VERSION=$version
+    fi
+  done <<< "$(echo "$ALL_AWS_FOR_FLUENT_BIT_VERSIONS")"
+}
+
+# Sets AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB variable with the latest version available in dockerhub.
+set_dockerhub_version() {
+  # Find the dockerhub version.
+  docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
+  tag_array=(`echo ${docker_hub_image_tags}`)
+  AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py ${tag_array[@]})
+}
+
+# Returns the regional endpoint.
+# For Standard partition, it would be "amazonaws.com"
+# For China partition, it would be "amazonaws.com.cn"
+get_regional_image_endpoint() {
+  region=${1}
+  endpoint="amazonaws.com"
+
+  if [ "${region}" = "cn-north-1" ] || [ "${region}" = "cn-northwest-1" ]; then
+  		endpoint="${endpoint}.cn"
+  fi
+  echo "${endpoint}"
+}
+
+#############################################################################################################
+# Functions to compare and validate image manifests
+#############################################################################################################
+
+# Checks if the image manifest for the provided image exists in the region.
+check_image_manifest_exists() {
+  manifest_image=${1}
+
+  # Inspect the manifest of the image.
+  # If the same doesn't exist, return 1.
+  image_manifest=$(docker manifest inspect "${manifest_image}" || echo "1")
+
+  if [ "${image_manifest}" = 1 ]; then
+    echo "No manifest found for ${manifest_image}"
+    exit 1
+  else
+    # If the image manifest exists, then ensure that the number of digests is equal to the supported Windows releases.
+    manifest_digests=$(echo "${image_manifest}" | jq -r '.manifests[].digest')
+    if [ $(echo "${manifest_digests}" | wc -l) != $SUPPORTED_WINDOWS_OS_RELEASES_NUMBER ]; then
+      echo "Incomplete manifest found for ${manifest_image}"
+      exit 1
+    fi
+    echo "Manifest validation completed for ${manifest_image}"
+  fi
+}
+
+# Compares image manifests of two images based on the image digests.
+compare_image_manifests() {
+  first_manifest_image=${1}
+  second_manifest_image=${2}
+
+  # Retrieve the manifests of both the images. If the same is not found then return 1.
+  first_manifest=$(docker manifest inspect "${first_manifest_image}" || echo "1")
+  second_manifest=$(docker manifest inspect "${second_manifest_image}" || echo "1")
+
+  if [ "${first_manifest}" = 1 ]; then
+    echo "No manifest found for ${first_manifest_image}"
+    return 1
+  elif [ "${second_manifest}" = 1 ]; then
+    echo "No manifest found for ${second_manifest_image}"
+    return 1
+  fi
+
+  # Obtain the digest entries from the manifest.
+  first_manifest_digests=$(echo "${first_manifest}" | jq -r '.manifests[].digest' | sort)
```

Comment:
This wont work on a manifest list right?

Reply:
So this method does compare manifest lists. `$(echo "${first_manifest}" | jq -r '.manifests[].digest' | sort)` returns a list of digests which we compare for both the images.